src/android/shakeDetector.java

```diff
@@ -0,0 +1,222 @@
+ */
+
+import de.appplant.cordova.plugin.localnotification.TriggerReceiver;
+import de.appplant.cordova.plugin.notification.Manager;
+
+/*
+ * Importing dependencies from the logger plugin
+ */
+import edu.berkeley.eecs.emission.BuildConfig;
+import edu.berkeley.eecs.emission.R;
+import edu.berkeley.eecs.emission.cordova.tracker.wrapper.SimpleLocation;
+import edu.berkeley.eecs.emission.cordova.tracker.wrapper.Transition;
+import edu.berkeley.eecs.emission.cordova.unifiedlogger.Log;
+import edu.berkeley.eecs.emission.cordova.usercache.UserCache;
+import edu.berkeley.eecs.emission.cordova.usercache.UserCacheFactory;
+
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+import android.os.Handler;
+import android.os.IBinder;
+
+import edu.berkeley.eecs.emission.cordova.tracker.Constants;
+import edu.berkeley.eecs.emission.cordova.unifiedlogger.Log;
+import edu.berkeley.eecs.emission.cordova.unifiedlogger.NotificationHelper;
+
+public class ShakeDetection extends Service implements SensorEventListener{
+    private SensorManager sensorManager;
+    private Sensor accelerometer;
+    private Sensor gyroscope;
+    private Sensor linearAccelerometer;
+    private float acceleration;
+    private float acceleration2;
+    private float currAcceleration;
+    private float lastAcceleration;
+    private long lastUpdated;
+
+    @Override
+    public IBinder onBind(Intent intent){
+        return null;
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId){
+        sensorManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
+        accelerometer = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
+        sensorManager.registerListener(this, accelerometer, SensorManager.SENSOR_DELAY_NORMAL, new Handler());
+        gyroscope = sensorManager.getDefaultSensor(Sensor.TYPE_GYROSCOPE);
+        sensorManager.registerListener(this, gyroscope, SensorManager.SENSOR_DELAY_NORMAL, new Handler());
+        linearAccelerometer = sensorManager.getDefaultSensor(Sensor.TYPE_LINEAR_ACCELERATION);
+        sensorManager.registerListener(this, linearAccelerometer, SensorManager.SENSOR_DELAY_NORMAL, new Handler());
+        return START_STICKY;
+    }
+
+    @Override
+    public  void onAccuracyChanged(Sensor sensor, int accuracy){
+    }
+
+    @Override
+    public void onSensorChanged(SensorEvent event){
+
+        long currTime = System.currentTimeMillis();
+        long deltaTime = currTime; //- lastUpdated;
+
+        //Only get readings if an "accident" has not occured within the past 0.5 seconds
+        if(deltaTime > 500) { //Make the 500 a const
+
+            if (event.sensor.getType() == Sensor.TYPE_ACCELEROMETER) {
+                float xAcceleration = event.values[0];
```

Comment:
if you stored values in the database, you could use a sequence of values to detect the shake instead of a single reading.